### DISCUSSION TITLE
文档上暂时没找到如何设置shape 为text-block字体的颜色和大小

### DISCUSSION THREAD

**Discussion by Momo23015258** [antvis/X6#1994] [Q&A] [answered]
文档上暂时没找到如何设置shape 为text-block字体的颜色和大小

**Comment by zrx1118** [comment]
this.graph.addNode({
      x: 200,
      y: 100,
      width: 100,
      height: 40,
      attrs: {
        body: {
          fill: 'none', // 填充
          stroke: "none" // 边框
        },
        label: {
          text: "Hello", // 文本
          fill: "#f00" //填充颜色 
        }
      }
    });
[文档](https://x6.antv.vision/zh/docs/tutorial/basic/node#%E5%AE%9A%E5%88%B6%E6%A0%B7%E5%BC%8F-attrs)

**Comment by Momo23015258** [comment]
这个设置对text-block的node不起作用，对rect节点起作用的



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年4月11日(周一) 中午11:44
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [antvis/X6] 文档上暂时没找到如何设置shape 为text-block字体的颜色和大小 (Discussion #1994)




 
this.graph.addNode({
 x: 200,
 y: 100,
 width: 100,
 height: 40,
 attrs: {
 body: {
 fill: 'none', // 填充
 stroke: "none" // 边框
 },
 label: {
 text: "Hello", // 文本
 fill: "#f00" //填充颜色
 }
 }
 });
 文档

**Comment by NewByVector** [comment, marked as answer]
使用 `label` 是文本容器的选择器。
```ts
graph.addNode({
  x: 160,
  y: 120,
  width: 360,
  height: 120,
  shape: 'text-block',
  text: `There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.`,
  attrs: {
    label: {
      style: {
        color: 'red'
      }
    }
  },
})
```

**Comment by Momo23015258** [comment]
好的多谢



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2022年4月11日(周一) 下午2:16
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [antvis/X6] 文档上暂时没找到如何设置shape 为text-block字体的颜色和大小 (Discussion #1994)




 
使用 label 是文本容器的选择器。
 graph.addNode({   x: 160,   y: 120,   width: 360,   height: 120,   shape: 'text-block',   text: `There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable.`,   attrs: {     label: {       style: {         color: 'red'       }     }   }, })
